AIFoV/src/main/java/dk/gruppeseks/aifov/AIFoVProvider.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package dk.gruppeseks.aifov;
+
+import dk.gruppeseks.bodtrd.common.data.Entity;
+import dk.gruppeseks.bodtrd.common.data.Path;
+import dk.gruppeseks.bodtrd.common.data.World;
+import dk.gruppeseks.bodtrd.common.data.entityelements.AIData;
+import dk.gruppeseks.bodtrd.common.data.entityelements.Body;
+import dk.gruppeseks.bodtrd.common.data.entityelements.Position;
+import dk.gruppeseks.bodtrd.common.data.util.Vector2;
+import dk.gruppeseks.bodtrd.common.exceptions.NoPathException;
+import dk.gruppeseks.bodtrd.common.services.AISPI;
+import java.util.ArrayList;
+import java.util.List;
+import org.openide.util.lookup.ServiceProvider;
+
+/**
+ *
+ * @author Morten
+ */
+@ServiceProvider(service = AISPI.class)
+public class AIFoVProvider implements AISPI
+{
+    private World _world;
+    private final int RAY_COUNT = 100;
+    private final int VISION_DISTANCE = 350;
+    private final double VISION_DEGREES = 90;
+    private final int DUMB_AI_RANGE = 80;
+
+    @Override
+    public Path getPath(Entity entity, Entity target, World world) throws NoPathException
+    {
+        List<Position> path = new ArrayList<>();
+        _world = world;
+        Position targetPos = rayCast(entity, target); // Providing the point on the circle it collided would result in better AI.
+
+        if (targetPos == null)
+        {
+
+        }
+        else
+        {
+            Vector2 distance = new Vector2(entity.get(Position.class), target.get(Position.class));
+            if (distance.getMagnitude() < DUMB_AI_RANGE)
+            {
+                path.add(new Position(target.get(Position.class))); // doesnt work as intended
```

Comment:
If distance is within dumb AI range (80 pixels) it doesn't follow the player properly (This is what isnt working as intended). A reference isnt parsed because that would result in it knowing where the player is if the player goes behind a wall.